spyglass/src/main/java/com/linkedin/android/spyglass/tokenization/impl/WordTokenizer.java

```diff
@@ -376,12 +376,23 @@ protected int getSearchStartIndex(final @NonNull Spanned text, int cursor) {
         }
 
         // Get index of the end of the last span before the cursor (or 0 if does not exist)
-        MentionSpan[] spans = text.getSpans(0, text.length(), MentionSpan.class);
-        int closestToCursor = 0;
-        for (MentionSpan span : spans) {
-            int end = text.getSpanEnd(span);
-            if (end > closestToCursor && end <= cursor) {
-                closestToCursor = end;
+        int closestLastSpanEnd = 0;
+        // iterate over all spans before the cursor
+        // we do this by finding the next span transition and looking back to find the closest span to the cursor
+        int nextSpanStart;
+        for (int searchStartIndex = 0; searchStartIndex < cursor; searchStartIndex = nextSpanStart) {
+
+            // find the next span transition
+            nextSpanStart = text.nextSpanTransition(searchStartIndex, text.length(), MentionSpan.class);
+
+            // get the spans from searchStartIndex to nextSpanStart
+            // of these, we find the closest span to the cursor
+            MentionSpan[] closestLastSpans = text.getSpans(searchStartIndex, nextSpanStart, MentionSpan.class);
+            for (MentionSpan span : closestLastSpans) {
+                int end = text.getSpanEnd(span);
+                if (end > closestLastSpanEnd && end <= cursor) {
+                    closestLastSpanEnd = end;
+                }
             }
```

Comment:
Not sure that this is actually faster given that the new code has more calls to get the spans for different substrings and to compute next span transitions (which internally will loop through all the spans, e.g. see [this](https://android.googlesource.com/platform/frameworks/base/+/refs/heads/main/core/java/android/text/SpannableStringInternal.java#341)).

It is also worth noting that the new code is a fair bit longer/more complicated.

 I'd like to understand the context for changing this. What is the reason for these changes? Are you running into measurable performance issues in your app? Have you benchmarked these methods? More info would be helpful. If we're going to make the code more complicated, we need to understand what we're getting in return.